packages/compass-collection/src/components/mock-data-generator-modal/script-generation-utils.spec.ts

```diff
@@ -0,0 +1,1309 @@
+import { expect } from 'chai';
+import { faker } from '@faker-js/faker/locale/en';
+import { generateScript, type FieldMapping } from './script-generation-utils';
+
+/**
+ * Helper function to test that generated document code is executable
+ *
+ * This function takes a complete mongosh script and extracts just the document
+ * generation logic to test it in isolation with the real faker.js library.
+ *
+ * @param script - Complete mongosh script containing a generateDocument function
+ * @returns The generated document object (for any possible extra validation)
+ */
+function testDocumentCodeExecution(script: string): any {
+  // The script contains: "function generateDocument() { return { ... }; }"
+  // The "{ ... }" part is the document structure
+
+  // Extract the return statement from the generateDocument function
+  const returnMatch = script.match(/return ([\s\S]*?);[\s]*\}/);
```

Comment:
Nit, completely optional, but this can be simplified a bit:

```suggestion
  const returnMatch = script.match(/return (.*?);\s*\}/s);
```